### DISCUSSION TITLE
Feature Request: LINQ syntactic sugar delegates (ex: orderby delegate )

### DISCUSSION THREAD

**Discussion by jwize** [dotnet/csharplang#2402] [Language Ideas]
When I write my queries I prefer to write most of my queries using LINQ query syntax so it would be nice to be able to create delegates that could be used delegates passed to the syntax.
```csharp
    // example 1:
    public static orderby IQuerybable<Category> DescriptionThenGroup(this IQueryable<Category> 
         query) 
    {  
          return query.OrderBy(c => c.Description).ThenBy(c => c.Group);
    }
    var categories = 
          from cat in db.Categories
          where cat.Description = "Siamese"
          orderby CategoryThenGroup   
          select cat;
````
```csharp  
    public static where IQuerybable<Employee> InCompany(this IQueryable<Employee> query, Int32 
    companyId) 
    { 
          return query.Where(e => e.CompanyId == companyId);
    }
    var employees =
          from e in db.Employees
          where InCompany(123)
          select e;
```

Please and thank  you!

**Comment by svick** [comment]
I agree that composability of `IQuerybale`-based LINQ is lacking, but I'm not sure this proposal would improve the situation much, at least when compared with [LinqKit](https://www.nuget.org/packages/LINQKit.Core/). Using LinqKit, your examples become:

```c#
public static readonly Expression<Func<Category, object>> DescriptionThenGroup =
    cat => new { cat.Description, cat.Group };

var categories = 
    from cat in db.Categories.AsExpandable()
    where cat.Description == "<cat>"
    orderby DescriptionThenGroup.Invoke(cat)
    select cat;
```

```c#
public static readonly Expression<Func<Employee, int, bool>> InCompany =
    (e, companyId) => e.CompanyId == companyId;

var employees =
    from e in db.Employees.AsExpandable()
    where InCompany.Invoke(e, currentCompany)
    select e;
```

I feel like this approach is more versatile than what you propose. Though I wouldn't mind if LinqKit became better integrated with the framework or language.

**Comment by jwize** [comment]
Regarding:  "I feel like this approach is more versatile than what you propose. Though I wouldn't mind if LinqKit became better integrated with the framework or language."

I don't want to be loud but the code you presented lacks expressiveness.  I have no idea what the code is trying to do anymore. It is confusing and looks like some class library code rather than a query for some data. A quick peek at LinqKit proves to be more of the same and I would much rather use LINQ joins than some of the proposed solutions given at the website link.
